criu/files-reg.c

```diff
@@ -392,6 +419,9 @@ static int create_ghost(struct ghost_file *gf, GhostFileEntry *gfe, struct cr_im
 	} else if (S_ISDIR(gfe->mode)) {
 		if ((ret = mkdirpat(AT_FDCWD, path, gfe->mode)) < 0)
 			msg = "Can't make ghost dir";
+	} else if (S_ISLNK(gfe->mode)) {
+		if ((ret = mklnk_ghost(path, gfe)) < 0)
+			msg = "Can't create ghost symlink";
```

Comment:
we probably want to be strict in images and report an error if
`!S_ISLNK(gfe->mode) && gfe->has_symlnk_target`

Reply:
> we probably want to be strict in images and report an error if
> `!S_ISLNK(gfe->mode) && gfe->has_symlnk_target`

I'm not sure, how to do that better way. If we want to be more strict with images, it also makes sense to refactor `create_ghost` function at all. There are also `gfe->has_rdev` parameter for instance.

Reply:
Right, up to you - I don't insist